src/main/java/io/vavr/control/Try.java

```diff
@@ -694,6 +694,26 @@ public final boolean isSingleValued() {
         return isSuccess() ? Iterator.of(get()) : Iterator.empty();
     }
 
+    /**
+     * If this is a Failure use the {@code mapFailure} to generate a new Try, from the Throwable error, which could be Success or Failure.
+     * If this is a Success it behaves as {@link Try#map(Function)}.
+     *
+     * @param mapFailure A recovery function taking a Throwable
+     * @param mapSuccess A mapping function taking the current value
+     * @param <U> The new component type
+     * @return
+     */
+    public final <U> Try<U> map(Function<? super Throwable, ? extends U> mapFailure, Function<? super T, ? extends U> mapSuccess) {
+        Objects.requireNonNull(mapFailure, "mapFailure is null");
+        Objects.requireNonNull(mapSuccess, "mapSuccess is null");
+
+        if (isFailure()) {
+            return Try.of(() -> mapFailure.apply(getCause()));
```

Comment:
Map with both parameters for a Try could have two behaviours. In the first one the mapFailure parameter acts as a recover function, while in the second one mapFailure acts as a mapping from Throwable to ? extends Throwable. 

In the second case, the behaviour will be the same as calling `toEither().map(mapFailure, mapSuccess)`, and, at the same time, we can ensure that the left contains only a Throwable value. Instead in the first scenario calling `toEither().map(mapFailure, mapSuccess)` feels like having an Either<Recover_Type, Success_Type>.

What do you think about it?

Reply:
Hi Theodor, that‘s a good question. I‘m currently sitting in a meetup and will take a look later. But I see parallels to the existing fold() and transform() methods. Maybe the map() methods are kind of redundant here...

Reply:
@thadumi I've double-checked the 2.0.0 branch, where I already hardened the upcoming API.

Several methods will change, especially [`transform`](https://github.com/vavr-io/vavr/blob/v2.0.0/src/main/java/io/vavr/control/Try.java#L701-L711):

```java
// Vavr 1.0
<U> U transform(Function<Try<T>, U> f)

// Vavr 2.0
<U> Try<U> transform(CheckedFunction<Throwable, Try<U>> ifFailure, CheckedFunction<T, Try<U>> ifSuccess)
```

`fold` is already there and will stay that way:

```java
<U> U fold(Function<Throwable, U> ifFailure, Function<T, U> ifSuccess)
```

By using `fold` and `transform` we are already able to map the underlying values:

```java
transform(err => Failure(new Error("")), x => Success("whatever"))
```

When Java Records & Switch Expressions & Pattern Matching & Object Deconstruction will be available, a convenient way in Java would be:

```java
var mappedTry = switch(myTry) {
    case Success(val) -> mapSuccess.apply(val)
    case Failure(err) -> mapFailure.apply(err)
}
```

I know that these solutions are a bit more verbose than the map overload we thought about. But it is important to me that we keep the API surface area small. There are already several options to achieve a similar result.

So let's keep the new `peek` and abandon the `map` overloads. Okay?

Reply:
Yes, I see your point. 
I just removed them.